From eeafd5d5bec1e323d34a998655686d096e5b2ecd Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Mon, 13 Jun 2016 14:36:08 +0800
Subject: [PATCH 1090/5242] MLK-12898: ov5640 mipi: Add more delay to wait
 sensor stable

commit  9c207e9fa0fa1ed868b085bfb0b0120e1d5c10ab from
https://source.codeaurora.org/external/imx/linux-imx.git

Add more delay to wait sensor stable.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
(cherry picked from commit c1d7c35b6d2c8b6ec69b90bac6febf673d04acc5)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../media/platform/mxc/capture/ov5640_mipi_v2.c    |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c b/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c
index ca85032..4ea4f6a 100644
--- a/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c
+++ b/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c
@@ -1106,12 +1106,9 @@ static int ov5640_init_mode(enum ov5640_frame_rate frame_rate,
 		/* dump the first two frames: 1/7.5*2
 		 * the frame rate of QSXGA is 7.5fps */
 		msec_wait4stable = 267;
-	} else if (frame_rate == ov5640_15_fps) {
-		/* dump the first nine frames: 1/15*9 */
+	} else {
+		/* dump the first eighteen frames: 1/30*18 */
 		msec_wait4stable = 600;
-	} else if (frame_rate == ov5640_30_fps) {
-		/* dump the first nine frames: 1/30*9 */
-		msec_wait4stable = 300;
 	}
 	msleep(msec_wait4stable);
 
-- 
1.7.9.5

